src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -231,47 +231,48 @@ function WebGLShadowMap( _renderer, _objects, maxTextureSize ) {
 
 	}
 
-	function getDepthMaterialVariant( useMorphing, useSkinning, useInstancing ) {
+	function getDepthMaterialVariant( useMorphing, useSkinning, useInstancing, alphaTest, alphaMap, map ) {
 
-		const index = useMorphing << 0 | useSkinning << 1 | useInstancing << 2;
+		const key = ( useMorphing << 0 | useSkinning << 1 | useInstancing << 2 | ( !! alphaMap ) << 3 | ( !! map ) << 4 ) + ',' + alphaTest;
```

Comment:
`alphaTest` is normally set only once. Especially since changing it requires a new shader compilation. However, if a user modulates `alphaTest`, this could produce a lot of depth material variants (since you need a unique one per `alphaTest` value).

One possible solution I can think of right now: Having a depth and/or distance material per material. In this way, you could just copy over the material properties and let the renderer handle a possible shader recompilation.